src/guide/project-maintenance.md

```diff
@@ -0,0 +1,66 @@
+
+~~~
+git clone git@github.com:OpenDuke/my-project.git project-merge
+cd project-merge
+git remote add upstream git@github.com:openjdk/my-project.git
+git remote add mainline git@github.com:openjdk/jdk.git
+~~~
+
+We clone the personal fork into a local directory, here called `project-merge`. We then set up two remotes, `upstream` and `mainline`.
+
+### Performing the merge
+
+The clone we set up above is used each time you want to bring changes from mainline in to your project. This is done by first pulling the changes from mainline and then pushing to your personal fork. A regular PR will then be created which you can integrate into your main project repository. It sounds easy, and it is, but there are a few details below to keep in mind.
+
+~~~
+cd project-merge
+git pull upstream master
+git push
+git switch -c Merge_mainline
+git fetch mainline
+git merge mainline/master
+~~~
+
+We start by updating the local fork with the latest changes from the main project repository. Note that we then create a new branch in which the merge will happen.
+
+The commands above will likely run without a hitch up until the `git merge`. This is where you need to combine the changes that were made in mainline with the changes that have been made in your project repository. If there are no conflicts you're in luck, then the merge will be completely automated and you will end up with a commit ready to push. If there are conflicts however you'll need to manually go through the files where the conflicts are and make sure you select the correct version for each change. Using `git status` you can see what files that need to be merged. Depending on how much code your project has touched this can be quite a bit of work.
+
+Once you have a working version of your merged code (make sure you test it properly) you're ready to create the commit and push. Please note that `git commit` is only needed if there were conflicts. If the changes were successfully merged by `git merge`, you're already ready to push.
+
+~~~
+git commit -m "Merge"
+git push --set-upstream origin Merge_mainline
+~~~
+
+Now it's time to create the PR on GitHub. Just opening the PR page in your browser will most often be enough to see a message about new code that was pushed to your personal fork, with a question if you want to create a PR. You do.
+
+Make sure the name of the PR starts with "Merge". You may have noticed that when you integrate a "normal" PR into an OpenJDK repository, all commits that has been done in that PR will be squashed into a single commit. For normal changes this is a good thing as each PR normally correspond to a single JBS issue, but for a merge it would be highly undesirable to squash all the different commits that you pull in from mainline. A PR with a name that starts with "Merge" won't be squashed. That means that all the changes that you brought over will remain separate changes.
```

Comment:
```suggestion
Make sure the name of the PR starts with "Merge". You may have noticed that when you integrate a "normal" PR into an OpenJDK repository, all commits that have been done in that PR will be squashed into a single commit. For normal changes this is a good thing as each PR normally correspond to a single JBS issue, but for a merge it would be highly undesirable to squash all the different commits that you pull in from mainline. A PR with a name that starts with "Merge" won't be squashed. That means that all the changes that you brought over will remain separate changes.
```